source/Tests/Infrastructure/OutgoingMessages/RSM012/AssertMeteredDataForMeteringPointEbixDocument.cs

```diff
@@ -0,0 +1,297 @@
+        return this;
+    }
+
+    public IAssertMeteredDataForMeteringPointDocumentDocument HasMeteringPointNumber(int seriesIndex, string expectedMeteringPointNumber, string expectedSchemeCode)
+    {
+        _documentAsserter.HasValue($"PayloadEnergyTimeSeries[{seriesIndex}]/MeteringPointDomainLocation/Identification", expectedMeteringPointNumber);
+        return this;
+    }
+
+    public IAssertMeteredDataForMeteringPointDocumentDocument HasMeteringPointType(int seriesIndex, MeteringPointType expectedMeteringPointType)
+    {
+        _documentAsserter.HasValue($"PayloadEnergyTimeSeries[{seriesIndex}]/DetailMeasurementMeteringPointCharacteristic/TypeOfMeteringPoint", expectedMeteringPointType.Code);
+        return this;
+    }
+
+    // OriginalTransactionIdReferenceId not present in ebix.
+    public IAssertMeteredDataForMeteringPointDocumentDocument HasOriginalTransactionIdReferenceId(int seriesIndex, string? expectedOriginalTransactionIdReferenceId)
+    {
+        return this;
+    }
+
+    // Not sure how this looks in ebix?
+    public IAssertMeteredDataForMeteringPointDocumentDocument HasOutDomain(int seriesIndex, string? expectedOutDomain)
+    {
+        if (expectedOutDomain is null)
+        {
+            _documentAsserter.IsNotPresent($"PayloadEnergyTimeSeries[{seriesIndex}]/OutDomain");
+            return this;
+        }
+
+        _documentAsserter.HasValue($"PayloadEnergyTimeSeries[{seriesIndex}]/OutDomain", expectedOutDomain);
+        return this;
+    }
+
+    public IAssertMeteredDataForMeteringPointDocumentDocument HasPoints(int seriesIndex, IReadOnlyList<AssertPointDocumentFieldsInput> expectedPoints)
+    {
+        var pointsInDocument = _documentAsserter
+            .GetElements($"PayloadEnergyTimeSeries[{seriesIndex}]/IntervalEnergyObservation");
+
+        pointsInDocument.Should().HaveSameCount(expectedPoints);
+
+        for (var i = 0; i < expectedPoints.Count; i++)
+        {
+            var (requiredPointDocumentFields, optionalPointDocumentFields) = expectedPoints[i];
+
+            _documentAsserter
+                .HasValue(
+                    $"PayloadEnergyTimeSeries[{seriesIndex}]/IntervalEnergyObservation[{i + 1}]/Position",
+                    requiredPointDocumentFields.Position.ToString());
+
+            if (optionalPointDocumentFields.Quantity.HasValue)
+            {
+                _documentAsserter
+                    .HasValue(
+                        $"PayloadEnergyTimeSeries[{seriesIndex}]/IntervalEnergyObservation[{i + 1}]/EnergyQuantity",
+                        optionalPointDocumentFields.Quantity.Value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                AssertElementNotPresent($"PayloadEnergyTimeSeries[{seriesIndex}]/IntervalEnergyObservation[{i + 1}]/EnergyQuantity");
+            }
+
+            if (optionalPointDocumentFields.Quality != null)
```

Comment:
```suggestion
            if (optionalPointDocumentFields.Quality is not null)
```
Align with "quality is not null" down below